src/Database/Type/DecimalType.php

```diff
@@ -100,10 +101,34 @@ public function toDatabase($value, Driver $driver)
     public function toPHP($value, Driver $driver)
     {
         if ($value === null) {
-            return null;
+            return $value;
+        }
+
+        // Using coercion is faster than casting
+        // @codingStandardsIgnoreStart
+        return (float)+$value;
```

Comment:
With a script similar to yours. But you are not testing anything there, the $i should be string, not an integer.

Reply:
I read PHP source code and now I understand what is happening in PHP.

If scalars are evaluated as `IS_LONG` (e.g. `'123'`), `(float)+$value` is a little faster than `(float)$value`:
```php
$t = microtime(true);
for ($i = 0; $i < 10000000; ++$i) {
    $value = (string)$i;
    (float)$value;
}
var_dump( microtime(true) - $t );
// float(1.2263522148132)

$t = microtime(true);
for ($i = 0; $i < 10000000; ++$i) {
    $value = (string)$i;
    (float)+$value;
}
var_dump( microtime(true) - $t );
// float(1.1857590675354)
```
Becase the latter doesn't call the `zend_strtod()` and does [a simple operation](https://github.com/php/php-src/blob/414aaa207cc59c279d3fd5ed83f6b92890e57023/Zend/zend_operators.c#L3000) to get long values in such cases
 
However, scalars are evaluated as `IS_DOUBLE` (e.g. `'123.0'`), `(float)+$value` is much slower than `(float)$value`:
```php
$t = microtime(true);
for ($i = 0; $i < 10000000; ++$i) {
    $value = $i . '.0';
    (float)$value;
}
var_dump( microtime(true) - $t );
// float(1.5314469337463)

$t = microtime(true);
for ($i = 0; $i < 10000000; ++$i) {
    $value = $i . '.0';
    (float)+$value;
}
var_dump( microtime(true) - $t );
// float(1.9330399036407)
```

Since float/decimal columns contain decimal points usually, I don't think this can be an optimization.

Could you please confirm the above two benchmarks?

Reply:
I can confirm your numbers, it seems I only checked this for ints, and not floats. Thanks for catching this!